### DISCUSSION TITLE
Bubibase DB filter current Day/Month

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#6270] [Help]
Hopefully this is me nearing the end of bombarding questions...

As mentioned in other questions I have opted to use Budibase DB as opposed to my own MYSQL database as it seems to have more flexibility

Although I have noticed that I cannot run DB Queries to get the info that I am looking for.

I always have a dashboard which gives a snapshot of how well the business is doing.

I usually have 3 sets of data... 

- Financial Year
- Current Month
- Todays Figures

The financial year seems easy enough to obtain by filtering between 2 sets of dates

However, I cannot find a way to have current month or todays data shown.

Usually I would run a query like:
Current Month:
`SELECT *
FROM table
WHERE DATE_FORMAT(JobDate, '%Y-%m') = DATE_FORMAT(NOW(), '%Y-%m')`

Today:
`SELECT * 
FROM Table
WHERE DATE_FORMAT(JobDate, '%d') = DATE_FORMAT(NOW(), '%d')`

With the queries above I dont need to do anything, when the month changes, the data on the dashboard starts from zero same with the 'todays' records when the day changes.

Can this be achieved using Budibase DB?

**Comment by melohagan** [comment]
Hey @PaulMcF1987 

You would most likely need to use a JavaScript binding in a **static** <a href="https://docs.budibase.com/docs/formula">Formula</a> column.

For example getting the current month is easy enough: https://www.w3schools.com/jsref/jsref_getmonth.asp

**Reply by PaulMcF1987** [reply]
@melohagan I have done that to get the month name.

However, what I am wanting is something similar to 
`SELECT SUM(Sales) AS Sales, SUM(Spend) AS Spend, SUM(Profit) AS Profit FROM table WHERE DATE_FORMAT(JobDate, '%Y-%m') = DATE_FORMAT(NOW(), '%Y-%m')`

Where only information from the current month is displayed.

With what you have mentioned above, if i have read correctly, I would need to go in every month and change the filter to display the current month.
Where as If I had used a my MYSQL database and used the above query, the results would reset at the start of every month and produce a fresh set of results.
Similarly with the `today` section, it would reset at the beginning of every day.

The problem with the filter in the `create view` section is that it only looks to allow filtering for a specific day, meaning I would need to update the filter daily,
Or for a specified time period,

**Reply by PaulMcF1987** [reply]
![image](https://user-images.githubusercontent.com/75368133/173114464-6c2c6893-dfca-4ca9-b156-00c7fd81d52d.png)
The above image should give an idea of what Im looking to day.
The financial year is fine, I can create a filter between 2 dates and that will run for 1 year before needing updates.

The `Months Figures` should tally up for the month and then reset the figures to 0 at the end of the month and begin tallying again
Similarly, the `todays figures` column should tally up the days work and then reset to 0 for the next day.

The query I pasted above does that.
However, I cannot se any way to replicate this with the Budibase DB

**Reply by PaulMcF1987** [reply]
is there any way at all to write queries in the Budibase DB?
I feel as though, giving that functionality would make a massive difference.
For example the above pic I have shared, gives this months figures, but for analytics purposes, we also have a graph which shows all of the same info
like the below
![image](https://user-images.githubusercontent.com/75368133/173116939-1e982d12-b8b8-45be-9462-9f369aa56d36.png)

But using Budibase DB you can only run one calculation per table, So I would need 3 tables to get the information that is in the graph, also, the graph on Budibase only allows 1 data provider, so rather than one graph, I would need 3 graphs...

A simple query like the above would make everything a lot easier.

Surely there has to be a way to achieve this?

**Reply by melohagan** [reply]
The Budibase Internal DB is limited in scope for queries and relationships. It is a basic NoSQL database - CouchDB.

The closest thing to queries you can do is combine the formulas with the views: https://docs.budibase.com/docs/views

You can do a sum calculation in a view.
